mne/epochs.py

```diff
@@ -308,8 +318,64 @@ def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
             else:
                 self.drop_log = drop_log
             events = events[selected]
-            if len(np.unique(events[:, 0])) != len(events):
-                raise RuntimeError('Event time samples were not unique')
+            u_evs, u_idxs, counts = np.unique(events[:, 0],
+                                              return_index=True,
+                                              return_counts=True)
+            if len(u_evs) != len(events):
+                msg = 'Multiple event codes for single event times found.'
+                if event_repeated == 'error':
+                    raise RuntimeError('Event time samples were not unique')
+
+                elif event_repeated == 'drop':
+                    logger.info('{} Keeping the first occurrence and '
+                                'dropping all others.'.format(msg))
+                    events = events[u_idxs]
+
+                elif event_repeated == 'merge':
+                    logger.info('{} Creating a new event code to reflect these'
+                                'simultaneous events and updating event_id.'
+                                .format(msg))
+                    new_events = events.copy()
+                    non_u_evs = u_evs[counts > 1]
+                    for ev in non_u_evs:
+
+                        # indices at which the non-unique events happened
+                        idxs = (events[:, 0] == ev).nonzero()[0]
+
+                        # Figure out event_codes prior to events to be merged
+                        prior_codes = events[idxs, 1]
+                        new_prior = np.nan
+                        if len(np.unique(prior_codes)) == 1:
+                            new_prior = np.unique(prior_codes)[0]
+
+                        # Make an event_id for the merged event
+                        ev_codes = events[idxs, 2]
+                        new_event_id = '{}'.format(ev_codes)
```

Comment:
Should not merge the event code integers, but rather the `event_id` keys. It makes it so that if I have:
```
event_id = {'aud': 1, 'vis': 2}
```
And events
```
events = [[0, 0, 1], [0, 0, 2]]
```
then in `merge` mode we will automagically do:
```
events_id['aud/vis'] = 3
events = [[0, 0, 3]]
```
What you wrote would do (I think):
```
events_id['1/2'] = 3
```
which is not as nice.

What I wrote above is basically the first unit test I would have started with for TDD, and eventually would put into `Notes` to describe the behavior (along with what happens for `drop` as well).

Reply:
+100 ! @sappelhoff you are yet to discover the joys of TDD ;)

Reply:
makes sense, see https://github.com/mne-tools/mne-python/pull/6688/commits/0282eb249ab10ec6a05b3b315970ba39103c81a3